src/creator/core.rs

```diff
@@ -1,15 +1,20 @@
 use anyhow::Result;
 use async_trait::async_trait;
-
-/// Re-export task request types from ingress
-pub use crate::ingress::types::TaskRequest;
+use std::collections::HashMap;
 
 /// The main creator trait. Implementations produce a payload and a round number.
 #[async_trait]
 pub trait Creator: Send + Sync {
     /// Compute and return the payload bytes and associated round.
     async fn get_payload_and_round(&self) -> Result<(Vec<u8>, u64)>;
-}
 
-/// Type alias for boxed creators (enables runtime polymorphism)
-pub type BoxedCreator = Box<dyn Creator>;
+    /// Get task metadata as key-value pairs.
+    ///
+    /// These metadata fields are used in the wire protocol messages and are typically
+    /// specific to the use case being implemented (e.g., counter, voting, etc.).
+    /// The metadata is flexible and can contain any number of key-value pairs.
```

Comment:
task data should probably be more abstract than a hashmap 
unsure what the implications of deperecating the `Box` usage is